[Rex Koh Jun Kai] iP

rexkoh425 opened the pull request.

rexkoh425 committed:
Added greetings and bye

rexkoh425 committed:
Added code to echo user input

Reads user input and prints the input

rexkoh425 committed:
Added List class

1) List class stores user input.
2) Changed to switch case to handle user input

rexkoh425 committed:
Added markings when listing items

1) Allow users to mark and unmark items

rexkoh425 committed:
Added markings when listing items
Edited code to follow java coding style

rexkoh425 committed:
Revert "Added markings when listing items"

This reverts commit 91ab99035b76a0727f97670a438e133caca120dc.

rexkoh425 committed:
Edited code to follow java coding style

rexkoh425 committed:
Added level-4 code

ToDoList class was integrated in to Task class using static members

rexkoh425 committed:
Added Text UI testing

input.txt contains the sequence of inputs

EXPECTED.txt contains the expected output from the sequence of inputs

rexkoh425 committed:
Improved code quality

Improved on abstraction
Used constant variables with meaningful names rather than just magic numbers.

rexkoh425 committed:
Removed redundant initializer

glenda-1506 commented on src/main/java/TheThinker.java:
```diff
@@ -0,0 +1,74 @@
+    }
+
+    public static void printSeparation(){
+        System.out.println("____________________________________________________________");
```
You may consider using "_".repeat(30) to improve conciseness.

glenda-1506 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,61 @@
+        System.out.println("OK, I've marked this task as not done yet:");
+        System.out.printf("  [%c][ ] " + currentTask.getTaskDescription() + "\n" , currentTask.getTaskType());
+        currentTask.setMarkedAsDone(false);
+    }
```
You could consider consolidating into a single method to reduce code duplication because setAsDone() and setAsNotDone() is nearly identical.

raneeng replied:
Could setAsDone and setAsNotDone be combined into one method?

raneeng replied:
Could setAsDone and setAsNotDone be combined into one method?

glenda-1506 commented on src/main/java/UserInputParser.java:
```diff
@@ -0,0 +1,70 @@
+    public static final int LENGTH_OF_DEADLINE = 8;
+    public static final int LENGTH_OF_SLASH_FROM = 5;
+    public static final int LENGTH_OF_SLASH_TO = 3;
+    public static final int LENGTH_OF_SLASH_BY = 3;
```
Great use of magic numbers.

glenda-1506 reviewed (commented):
Overall great code, but you can try catching invalid inputs in your program.

raneeng commented on src/main/java/TheThinker.java:
```diff
@@ -0,0 +1,74 @@
+        default:
+            System.out.println("try again with the correct format");
+            break;
+        }
```
good switch statement indentation! :-)

raneeng commented on src/main/java/UserInputParser.java:
```diff
@@ -0,0 +1,70 @@
+        remainingTaskDescription = remainingTaskDescription.substring(indexOfFirstSlash + LENGTH_OF_SLASH_FROM).trim();
+        int indexOfSecondSlash = remainingTaskDescription.indexOf("/to");
+        String startTime = remainingTaskDescription.substring(0, indexOfSecondSlash).trim();
+        String endTime = remainingTaskDescription.substring(indexOfSecondSlash + LENGTH_OF_SLASH_TO).trim();
```
clear variable naming! :-)

raneeng commented on src/main/java/TheThinker.java:
```diff
@@ -0,0 +1,74 @@
+            String userAction = UserInputParser.parseUserAction();
+            doTaskAccordingToUserAction(userAction);
+
+        }while(!userInput.equals("bye"));
```
use consistent spacing,
eg:
in line 5 you have:
public static void main(String[] args) {
in line 18 you have:
public static void printGreeting(){

raneeng reviewed (commented):
Overall clear and logical, well done :-)

rexkoh425 committed:
added some error handling

Created a new checked exception

rexkoh425 committed:
added more error handling

Program does not allow empty fields like time , task now

rexkoh425 committed:
Merge branch 'branch-Level-5'

rexkoh425 committed:
Improved code readability

rexkoh425 committed:
Divided Classes into packages

Updated UI-text texting according to packages

rexkoh425 committed:
implemented delete task command

rexkoh425 committed:
implemented File class

File class able to get lines from a file

rexkoh425 committed:
added auto-write to file

when list changes , the new task list is written

rexkoh425 committed:
Added more error handling to file access

rexkoh425 committed:
Fixed error

rexkoh425 committed:
Merge branch 'branch-Level-6'

rexkoh425 committed:
Resolved merge conflicts

Resolved merge conflicts from level-7 -> master

rexkoh425 committed:
Added function lost due to merge

convertToFileFormat was added

rexkoh425 committed:
Create JAR manifest file

tzerbin commented on src/main/java/TheThinker/Command/TheThinker.java:
```diff
@@ -0,0 +1,139 @@
+package TheThinker.Command;
+import TheThinker.Exceptions.FormattingException;
```
Perhaps leave appropriate whitespace between wrapping package and import statements?
(This is part of intermediate coding guidelines on whitespace.)

tzerbin commented on src/main/java/TheThinker/Command/TheThinker.java:
```diff
@@ -0,0 +1,139 @@
+        NewFile data = null;
+
+        try {
+            data = new NewFile("TaskContents.txt");
```
There shouldn't be usage of magic strings and numbers in the entire codebase.

tzerbin commented on src/main/java/TheThinker/Command/TheThinker.java:
```diff
@@ -0,0 +1,139 @@
+    }
+
+    public static void printSeparation(){
+        System.out.println("_".repeat(60));
```
Very nice use of repeat() method!

tzerbin commented on src/main/java/TheThinker/NewFile/NewFile.java:
```diff
@@ -0,0 +1,89 @@
+    }
+
+    public void loadFile() throws FileNotFoundException {
+        Scanner SCANNER = new Scanner(this.file);
```
Please refrain from using SCREAMING_SNAKE_CASE for non-final variables.

tzerbin commented on src/main/java/TheThinker/NewFile/NewFile.java:
```diff
@@ -0,0 +1,89 @@
+        Task.listTasks();
+    }
+
+    public void doAccordingToParameters(String[] parameters){
```
Be more descriptive of the action that is being done in the method name.

tzerbin commented on src/main/java/TheThinker/Parser/UserInputParser.java:
```diff
@@ -0,0 +1,135 @@
+
+    public static Task parseTodo() throws FormattingException {
+
+        String taskType = userInput.substring(0 , LENGTH_OF_TODO);
```
Very nice descriptive names for variables in this file.

tzerbin reviewed (commented):
Very good attempt in following the guidelines.
Just some small issues here and there.

rexkoh425 committed:
Correct code based on comments

rexkoh425 committed:
Added Ui class

Ui class controls formatting commands in the terminal

rexkoh425 committed:
Split Task class to TaskList class

rexkoh425 committed:
Broken down code to more OOP

rexkoh425 committed:
Broken down code to more OOP

rexkoh425 committed:
Added date class

Users can now input using shortform dates such as 20/10/2019 1800 and will be converted to 20 October 2019 , 6 pm

rexkoh425 committed:
Added date filter command

rexkoh425 committed:
Merge pull request #1 from rexkoh425/branch-Level-8

Branch level 8

rexkoh425 committed:
Added keyword filter command

rexkoh425 committed:
Changed GET_FORMAT to FIND_FORMAT

rexkoh425 committed:
Merge pull request #2 from rexkoh425/Branch-Level-9

Branch level 9

rexkoh425 committed:
changed public to private methods

rexkoh425 committed:
added javadoc comments

Added for File package and main class

rexkoh425 committed:
edit existing javadoc comments

rexkoh425 committed:
update javadoc for Date class

rexkoh425 committed:
update javadoc for UserInputParser

rexkoh425 committed:
update javadoc for Task package

rexkoh425 committed:
update javadoc for Ui package

rexkoh425 committed:
Merge pull request #3 from rexkoh425/branch-A-JavaDoc

Branch a java doc

rexkoh425 committed:
Updated user guide

rexkoh425 committed:
User can run jar file anywhere within ip

rexkoh425 committed:
removed unnecessary imports

rexkoh425 committed:
build new jar file

rexkoh425 committed:
Added auto creating of folder and file , file path now is relative

rexkoh425 committed:
Allow user to choose whether to save file

rexkoh425 committed:
used better naming for functions

rexkoh425 committed:
added find to include other type of dates

rexkoh425 committed:
Added clear error messages

added possible inputs as test case in input.txt as well

rexkoh425 committed:
track TaskContents.txt

rexkoh425 committed:
Edit javadoc and improve code quality

rexkoh425 committed:
Improved user guide

rexkoh425 committed:
Commit for new jar file

rexkoh425 committed:
add jar file instructions

rexkoh425 committed:
checked code quality + UI testing

rexkoh425 committed:
update code quality

included <br> in user guide for number bullets too

rexkoh425 committed:
added <br> to UG for format

rexkoh425 committed:
renamed packages to lowercase

rexkoh425 committed:
added spacing to {

rexkoh425 committed:
update javadoc code quality

rexkoh425 committed:
update javadoc code quality

rexkoh425 committed:
updated ui testing

rexkoh425 committed:
rebuild jar file

rexkoh425 committed:
Ui testing fully passes now

Ui testing was failing because of trailing spaces

rexkoh425 committed:
Coding style corrected

rexkoh425 committed:
Removed checkstyle

rexkoh425 committed:
Corrected user guide error
